flutter-for-android.md

```diff
@@ -0,0 +1,1864 @@
+    );
+  }
+}
+
+class SampleAppPage extends StatefulWidget {
+  SampleAppPage({Key key}) : super(key: key);
+
+  @override
+  _SampleAppPageState createState() => new _SampleAppPageState();
+}
+
+class _SampleAppPageState extends State<SampleAppPage> {
+  List widgets = new List();
+
+  @override
+  void initState() {
+    super.initState();
+
+    loadData();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+        appBar: new AppBar(
+          title: new Text("Sample App"),
+        ),
+        body: new ListView.builder(
+            itemCount: widgets.length,
+            itemBuilder: (BuildContext context, int position) {
+              return getRow(position);
+            }));
+  }
+
+  Widget getRow(int i) {
+    return new Padding(padding: new EdgeInsets.all(10.0), child: new Text("Row ${widgets[i]["title"]}"));
+  }
+
+  loadData() async {
+    String dataURL = "https://jsonplaceholder.typicode.com/posts";
+    http.Response response = await http.get(dataURL);
+    setState(() {
+      widgets = JSON.decode(response.body);
+    });
+  }
+}
+{% endprettify %}
+
+
+
+## What is the equivalent of an AsyncTask or IntentService on Android
+
+In Android, when you want to access a Network resource you would
+typically spawn an AsyncTask that would run code outside of the UI thread
+to prevent your UI from blocking. The AsyncTask has a thread pool that manages
+the threads for you. 
+
+Since Flutter is single threaded and runs an event loop (like Node.js), you
+don't have to worry about thread management or spawing AsyncTasks or IntentServices.
+
+To run code asynchronously you can declare the function as an async function 
+and await on long running tasks in the function
+
+<!-- skip -->
+{% prettify dart %}
+loadData() async {
+  String dataURL = "https://jsonplaceholder.typicode.com/posts";
+  http.Response response = await http.get(dataURL);
+  setState(() {
+    widgets = JSON.decode(response.body);
+  });
+}
+{% endprettify %}
+
+This is how you would typically do network or database calls. 
+
+With regards to AsyncTask which follows a three step model onPreExecute, 
+doInBackground and onPostExecute, doing a similar model in Flutter is easier. 
+You can write non-blocking code as it is, like you would in onPreExecute 
+and onPostExecute and to run code asynchronously you would call an async 
+function.
+
+onPostExecute takes the value from the result of the doInBackground 
+task and updates the UI, in Flutter as you can see above you would take 
+the value returned from the await’ed function and update the UI by calling 
+setState.
+
+Yet, there are times where you may be processing a large amount of data and 
+your UI could hang. 
+
+Like AsyncTask is possible to take advantage of multiple CPU cores to do long 
+running or computationally intensive tasks. This is done by using Isolates.
+
+Isolates are a separate execution thread that runs and do not share any memory 
```

Comment:
s/thread/memory heap. ....... Isolates are implemented with threads under the covers.